operators/go/gen_subm_operator.sh

```diff
@@ -148,12 +148,66 @@ function export_subm_op() {
   cp -a $op_dir/. $op_out_dir/
 }
 
+function add_subm_cluster_crd_to_operator() {
+  pushd $op_dir
+  api_version=submariner.io/v1alpha1
+  kind=Cluster
+  operator-sdk add crd --api-version=$api_version --kind=$kind || true
```

Comment:
This command fails at first run. But immediately re-running it passes.

+ operator-sdk add crd --api-version=submariner.io/v1alpha1 --kind=Endpoint
INFO[0000] Generating Custom Resource Definition (CRD) version submariner.io/v1alpha1 for kind Endpoint. 
Error: crd scaffold failed: (error generating CRD for Group submariner Version v1alpha1 Kind Endpoint)
Usage:
  operator-sdk add crd [flags]

Flags:
      --api-version string   Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)
  -h, --help                 help for crd
      --kind string          Kubernetes CustomResourceDefintion kind. (e.g AppService)

Global Flags:
      --verbose   Enable verbose logging

+ true
+ operator-sdk add crd --api-version=submariner.io/v1alpha1 --kind=Endpoint
INFO[0000] Generating Custom Resource Definition (CRD) version submariner.io/v1alpha1 for kind Endpoint. 
INFO[0000] Created deploy/crds/submariner.io_v1alpha1_endpoint_cr.yaml 
INFO[0000] CRD generation complete.

Reply:
```
--- a/operators/go/gen_subm_operator.sh
+++ b/operators/go/gen_subm_operator.sh
@@ -177,7 +177,8 @@ function add_subm_endpoint_crd_to_operator() {
   pushd $op_dir
   api_version=submariner.io/v1alpha1
   kind=Endpoint
-  operator-sdk add crd --api-version=$api_version --kind=$kind || true
+  operator-sdk add crd --api-version=$api_version --kind=$kind --verbose || true
+  operator-sdk add crd --api-version=$api_version --kind=$kind --verbose
```

Yeah, I see the same behavior - strange. Fails the first time with no obvious error, then immediately works when run a second time.

```
+ operator-sdk add crd --api-version=submariner.io/v1alpha1 --kind=Endpoint --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Generating Custom Resource Definition (CRD) version submariner.io/v1alpha1 for kind Endpoint. 
Error: crd scaffold failed: (error generating CRD for Group submariner Version v1alpha1 Kind Endpoint)
Usage:
  operator-sdk add crd [flags]

Flags:
      --api-version string   Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)
  -h, --help                 help for crd
      --kind string          Kubernetes CustomResourceDefintion kind. (e.g AppService)

Global Flags:
      --verbose   Enable verbose logging

+ true
+ operator-sdk add crd --api-version=submariner.io/v1alpha1 --kind=Endpoint --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Generating Custom Resource Definition (CRD) version submariner.io/v1alpha1 for kind Endpoint. 
INFO[0000] Created deploy/crds/submariner.io_v1alpha1_endpoint_cr.yaml 
INFO[0000] CRD generation complete.                     
```

I'll try adding a sleep as Stephen suggested.

Reply:
Failed again in the same way, this time with a 10s sleep before the first one.

```
+ sleep 10
+ operator-sdk add crd --api-version=submariner.io/v1alpha1 --kind=Endpoint --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Generating Custom Resource Definition (CRD) version submariner.io/v1alpha1 for kind Endpoint. 
Error: crd scaffold failed: (error generating CRD for Group submariner Version v1alpha1 Kind Endpoint)
Usage:
  operator-sdk add crd [flags]

Flags:
      --api-version string   Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)
  -h, --help                 help for crd
      --kind string          Kubernetes CustomResourceDefintion kind. (e.g AppService)

Global Flags:
      --verbose   Enable verbose logging

+ true
+ operator-sdk add crd --api-version=submariner.io/v1alpha1 --kind=Endpoint --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Generating Custom Resource Definition (CRD) version submariner.io/v1alpha1 for kind Endpoint. 
INFO[0000] Created deploy/crds/submariner.io_v1alpha1_endpoint_cr.yaml 
INFO[0000] CRD generation complete.                     
```